### DISCUSSION TITLE
Warning: OPMODE Input Warning : The OPMODE 0110X0X with CARRYINSEL  000 to DSP48E1 instance is invalid.

### DISCUSSION THREAD

**Discussion by s095339** [bol-edu/HLS-SOC-Discussions#52] [Q&A]
Hello teacher and TA, I and working on Final project on HLS.
My subject is "Adaptive Beamforming", and I build a project to design and run the sub_kernel "QR Factorization".
Here is my project: https://github.com/s095339/Adaptive_Beamforming.
The host code file is  "host.cpp", and the related kernel file are "Top_Kernel.cpp","dut_type" and all files in ./hw

I copied the directory from the Vitis solver Library, which is located at ["Vitis_Library/solver/L2/tests/hw/geqrf"](https://github.com/Xilinx/Vitis_Libraries/tree/main/solver/L2/tests/hw/geqrf), and add the related  file to make sure all the dependency are included. 

I can run software emulation by typing "make run TARGET=sw_emu". Everything can run successfully.
But when I run  hardware emulation, the execution get stuck when running the host code. I think compiler can synthesis can compile my code successfully.
Here is the information generated on my mobaXterm :
![image](https://github.com/bol-edu/HLS-Discussions/assets/78518884/840d74bf-2c0e-4ecd-b639-4059fc98fba7)

I checked the log file that generated during the emulation:

![image](https://github.com/bol-edu/HLS-Discussions/assets/78518884/68bf19bb-6a0d-4ea4-b0e9-0d78c85217d9)
I found that it keeps generating the warning message

I have no idea why this message shows on the log file and what are the message refer to . it seems that the warning slows the emulation down. I hope anyone can help me solving the problem.

**Comment by boledulab** [comment]
Maybe you can try to disable the check 
https://blog.csdn.net/jerwey/article/details/108003919
